cypress/integration/findings_spec.js

```diff
@@ -3,21 +3,213 @@
+
+    // Continue to next page
+    cy.contains('Next').click({ force: true });
+
+    // Confirm page is reached
+    cy.contains('Review and create');
+
+    // Confirm field mappings registered
+    cy.contains('Field mapping');
+    for (let field in TEST_FIELD_MAPPINGS) {
+      const mappedTo = TEST_FIELD_MAPPINGS[field];
+
+      cy.contains(field);
+      cy.contains(mappedTo);
+    }
+
+    // Create the detector
+    cy.get('button').contains('Create').click({ force: true });
+
+    cy.wait(10000);
   });
 
-  //TODO - get findings to be present, then test
+  it('displays findings based on recently ingested data', () => {
+    // Visit Findings page
+    cy.visit(`${Cypress.env('opensearch_dashboards')}/app/${PLUGIN_NAME}#/findings`);
+    cy.wait(10000);
+
+    // Confirm arrival at Findings page
+    cy.contains('Last 15 minutes');
+    cy.contains('Show dates');
+
+    // Ingest a new document
+    cy.ingestDocument('cypress-test-windows', TEST_DOCUMENT);
+
+    // wait for detector interval to pass
+    cy.wait(60000);
 
-  it('displays findings based on recently ingested data', () => {});
+    // Click refresh
+    cy.get('button').contains('Refresh').click({ force: true });
+
+    // Check for non-empty findings list
+    cy.contains('No items found').should('not.exist');
+
+    // Check for expected findings
+    cy.contains('test detector');
+    cy.contains('Windows');
+    cy.contains('Low');
+  });
+
+  it('displays finding details flyout when user clicks on Finding ID or View details icon', () => {
+    // Click findingId to trigger Finding details flyout
+    cy.get(`[data-test-subj="findings-table-finding-id"]`).click();
+
+    // Confirm flyout contents
+    cy.contains('Finding details');
+    cy.contains('Rule details');
+    cy.contains('Search Findings').should('not.exist');
+
+    // Close Flyout
+    cy.get(`button[data-test-subj="close-finding-details-flyout"]`).click();
+
+    // wait for toasts to clear
+    cy.wait(10000);
+
+    // Click View details icon
+    cy.get(`[data-test-subj="view-details-icon"]`).click();
+
+    // Confirm flyout contents
+    cy.contains('Finding details');
+    cy.contains('Rule details');
+    cy.contains('Search Findings').should('not.exist');
+
+    // Open rule details for each rule
+    cy.contains('USB Device Plugged').click({ force: true });
+
+    // Confirm content
+    cy.contains('Documents');
+    cy.contains('Detects plugged USB devices');
+    cy.contains('Low');
+    cy.contains('Windows');
+    cy.contains('cypress-test-windows');
+  });
+
+  it('allows user to view more rule details upon clicking rule name', () => {
+    // Click rule link
+    cy.get('a')
+      .contains('USB Device Plugged')
+      .invoke('removeAttr', 'target')
+      .click({ force: true });
+
+    // Confirm destination reached
+    cy.contains('Rules');
+    cy.contains('Import rule');
+    cy.contains('Create new rule');
```

Comment:
Would it make more sense to make use of the [url](https://docs.cypress.io/api/commands/url#Syntax) function here to confirm we were redirected as expected?

Reply:
That is interesting, and I will look into that!

Reply:
done